packages/runtime/example/dialog/list.html

```diff
@@ -0,0 +1,671 @@
+                                },
+                              },
+                              wait: {
+                                type: 'delay',
+                                time: 1
+                              },
+                              disabled: 'false',
+                            },
+                          ],
+                        },
+                      },
+                      {
+                        type: 'core/v1/slot',
+                        properties: {
+                          container: {
+                            id: 'listItemTemplate-{{$listItem.id}}',
+                            slot: 'content',
+                          },
+                        },
+                      },
+                    ],
+                  },
+                ],
+              },
+              traits: [
+                {
+                  type: 'core/v1/slot',
+                  properties: {
+                    container: {
+                      id: 'root',
+                      slot: 'root',
+                    },
+                  },
+                },
+              ],
+            },
+            // edit dialog
+            {
+              id: 'editDialog',
+              type: 'chakra_ui/v1/dialog',
+              properties: {
+                title: 'This is a dialog',
+                confirmButton: {
+                  text: 'save',
+                  colorScheme: 'purple'
+                },
+                cancelButton: {
+                  text: 'cancel'
+                }
+              },
+              traits: [
+                {
+                  type: 'core/v1/slot',
+                  properties: {
+                    container: {
+                      id: 'root',
+                      slot: 'root',
+                    },
+                  },
+                },
+                // dialog events
+                {
+                  type: 'core/v1/event',
+                  parsedType: {
+                    version: 'core/v1',
+                    name: 'event',
+                  },
+                  properties: {
+                    events: [
+                      // when click confirm
+                      {
+                        event: 'confirmDialog',
+                        componentId: 'editDialog',
+                        method: {
+                          name: 'confirmDialog'
+                        },
+                        wait: {},
+                        disabled: 'false',
+                      },
+                      {
+                        event: 'confirmDialog',
+                        componentId: 'root',
+                        method: {
+                          name: 'setValue',
+                          parameters: {
+                            key: 'listData',
+                            value: `{{
+                              (function() {
```

Comment:
Although we support use function in expression, but it is not good pratice to write such complicated expression. It is hard to imagine user write long expression in mata-ui editor. We can divide it to 2 dialogs, edit and create dialog, to keep expression simple.

Reply:
ok